packages/terra-popup/src/Popup.jsx

```diff
@@ -0,0 +1,274 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Portal from 'react-portal';
+import PopupContent from './_PopupContent';
+import PopupArrow from './_PopupArrow';
+import PopupOverlay from './_PopupOverlay';
+import TetherComponent from './_TetherComponent';
+import { parseStringPair, isVerticalAttachment, primaryArrowPosition, primaryMarginStyle, switchAttachmentToRTL, mirrorAttachment, getContentOffset, arrowPositionFromBounds, leftOffset, topOffset } from './_PopupUtils';
+import './Popup.scss';
+
+const HEIGHT_KEYS = ['40', '80', '120', '160', '240', '320', '400', '480', '560', '640', '720', '800', '880'];
+const HEIGHT_VALUES = [40, 80, 120, 160, 240, 320, 400, 480, 560, 640, 720, 800, 880];
+
+const WIDTH_KEYS = ['160', '240', '320', '640', '960', '1280', '1760'];
+const WIDTH_VALUES = [160, 240, 320, 640, 960, 1280, 1760];
+
+const CREATE_MAPPING = ((keys, values) => {
+  const combinations = {};
+  for (let i = 0; i < keys.length; i += 1) {
+    combinations[keys[i]] = values[i];
+  }
+  return combinations;
+});
+
+const WIDTH_MAPPING = CREATE_MAPPING(WIDTH_KEYS, WIDTH_VALUES);
+const HEIGHT_MAPPING = CREATE_MAPPING(HEIGHT_KEYS, HEIGHT_VALUES);
+
+const propTypes = {
+  /**
+   * The children to be displayed as content within the popup.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Callback function indicating a close condition was met, should be combined with isOpen for state management.
+   */
+  onRequestClose: PropTypes.func.isRequired,
+  /**
+   * Target element for the popup to anchor to.
+   */
+  targetRef: PropTypes.func.isRequired,
+  /**
+   * Bounding container for the popup, will use window if no value provided.
```

Comment:
Matt once mentioned we want to avoid binding to window. @mjhenkes is the desired default?